scanner/windows.go

```diff
@@ -50,30 +52,60 @@ func newWindows(c config.ServerInfo) *windows {
 func detectWindows(c config.ServerInfo) (bool, osTypeInterface) {
 	tmp := c
 	tmp.Distro.Family = constant.Windows
-
-	if isLocalExec(c.Port, c.Host) {
-		if r, r2 := exec(tmp, `$CurrentVersion = (Get-ItemProperty -Path "Registry::HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion"); Format-List -InputObject $CurrentVersion -Property ProductName, CurrentVersion, CurrentMajorVersionNumber, CurrentMinorVersionNumber, CurrentBuildNumber, UBR, CSDVersion, EditionID, InstallationType`, noSudo), exec(tmp, `(Get-ItemProperty -Path "Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment").PROCESSOR_ARCHITECTURE`, noSudo); (r.isSuccess() && r.Stdout != "") && (r2.isSuccess() && r2.Stdout != "") {
-			w := newWindows(c)
-			osInfo, err := parseRegistry(r.Stdout, strings.TrimSpace(r2.Stdout))
-			if err != nil {
-				w.setErrs([]error{xerrors.Errorf("Failed to parse Registry. err: %w", err)})
-				return true, w
+	w := newWindows(tmp)
+	w.shell = func() string {
+		if r := w.exec("echo $env:OS", noSudo); r.isSuccess() {
+			switch strings.TrimSpace(r.Stdout) {
+			case "$env:OS":
+				return "cmd.exe"
+			case "Windows_NT":
+				return "powershell"
+			default:
+				if rr := w.exec("Get-ChildItem env:OS", noSudo); rr.isSuccess() {
+					return "powershell"
+				}
+				return "unknown"
 			}
+		}
+		return "unknown"
```

Comment:
Is it good to proceed in this case? Hmm... Better to just exit with error message of ssh?

Reply:
It's good. No treatment needed!